auto_dev/data/templates/compose/tendermint.yaml

```diff
@@ -12,15 +12,47 @@ services:
     network_mode: ${NETWORK_MODE}
     environment:
       - ID=0
-      - PROXY_APP=tcp://${HOST_NAME}
       - TMHOME=/tendermint/node0
       - CREATE_EMPTY_BLOCKS=true
       - DEV_MODE=0
       - LOG_FILE=node_0.txt
       - LOG_LEVEL=INFO
       - WRITE_TO_LOG=true
-    working_dir: /tendermint
-    command: ["run", "--no-reload", "--host=0.0.0.0", "--port=8080"]
+      - HOST_NAME=${HOST_NAME}
+    working_dir: /app
+    entrypoint: []  # override image's "/usr/bin/flask" entrypoint
+    command:
+      - /bin/sh
+      - -c
+      - |
+        # Try each port using Python socket binding
+        python3 - << 'PYEOF' > /tmp/port.txt
+        import socket
+        import errno
+        ports = range(8080, 8096)  # 8080 through 8095
+        for p in ports:
+            try:
+                s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                s.bind(('127.0.0.1', p))
+                s.close()
+                print(p)  # This will be captured in port.txt
+                break
+            except socket.error as e:
+                if e.errno != errno.EADDRINUSE:
+                    print(f"Port {p} error: {e}")
+        PYEOF
+
+        port=$$(cat /tmp/port.txt)
+        if [ -z "$$port" ]; then
+            echo "No ports available"
+            exit 1
+        fi
+
+        echo "Found port $$port, starting Flask..."
+        export FLASK_APP=app:create_server
+        export PROXY_APP="tcp://${HOST_NAME}"
+        echo "PROXY_APP=$$PROXY_APP"
+        flask run --no-reload --host=0.0.0.0 --port=$$port
```

Comment:
I don't think it's good practice to modify the docker file like this. 
Especially as all you actually want to do is change some env vars

Please check the docker compose syntax for passing env vars
I.e. exposed port right?
Also this new port needs to be bubbled up to the run layer to be given to the agent so that it knows what port to talk to

Reply:
okay I set FLASK_PORT in run.py and pass that to the dockerfile